src/app/components/client/ExposureCardDataClass.tsx

```diff
@@ -4,97 +4,143 @@
 
 "use client";
 
-import { ReactElement } from "react";
+import { ReactElement, useEffect, useState } from "react";
 import { OnerepScanResultRow } from "knex/types/tables";
 import styles from "./ExposureCard.module.scss";
-import { SubscriberBreach } from "../../../utils/subscriberBreaches";
 import { useL10n } from "../../hooks/l10n";
+import { Exposure, isScanResult } from "./ExposureCard";
+import { StateAbbr } from "../../../utils/states";
+import { HibpBreachDataTypes } from "../../(nextjs_migration)/(authenticated)/user/breaches/breaches";
 
-type DataBrokerDataClassProps = {
-  scanResultData: OnerepScanResultRow;
-  exposureCategoryLabel: string;
-  num: number;
-  icon: ReactElement;
-  isPremiumUser: boolean;
-};
+type OnerepScanResultSerializedColumns = Extract<
+  keyof OnerepScanResultRow,
+  "emails" | "phones" | "addresses" | "relatives"
+>;
 
-export const DataBrokerDataClass = (props: DataBrokerDataClassProps) => {
-  const emailsList = props.isPremiumUser ? (
-    <ul className={styles.emailsList}>
-      {props.scanResultData.emails.map((email: string, index: number) => (
-        <li key={index}>{email}</li>
-      ))}
-    </ul>
-  ) : (
-    <></>
-  );
-
-  return (
-    <ExposureCardDataClassLayout
-      icon={props.icon}
-      label={props.exposureCategoryLabel}
-      count={props.num}
-      emailData={emailsList}
-    />
-  );
+type DataBrokerAddress = {
+  city: string;
+  state: StateAbbr;
+  street?: string;
+  zip?: string;
 };
 
-type BreachDataClassProps = {
-  subscriberBreachData: SubscriberBreach;
-  exposureCategoryLabel: string;
-  icon: ReactElement;
+type PremiumDataClassDetailsProps = {
+  exposure: Exposure;
+  dataBrokerDataType: OnerepScanResultSerializedColumns;
 };
 
-export const BreachDataClass = (props: BreachDataClassProps) => {
-  const emailLength = props.subscriberBreachData.emailsAffected.length;
-
-  const emailsList = (
-    <ul className={styles.emailsList}>
-      {props.subscriberBreachData.emailsAffected.map(
-        (email: string, index: number) => (
-          <li key={index}>{email}</li>
-        ),
-      )}
-    </ul>
-  );
+// Only for data broker cards
+const PremiumDataClassDetails = (props: PremiumDataClassDetailsProps) => {
+  const { exposure, dataBrokerDataType } = props;
+  let content: JSX.Element[] | null = null;
 
-  return (
-    <ExposureCardDataClassLayout
-      icon={props.icon}
-      label={props.exposureCategoryLabel}
-      count={emailLength}
-      emailData={emailsList}
-    />
-  );
+  if (isScanResult(exposure)) {
+    const addresses = exposure.addresses || [];
+
+    switch (dataBrokerDataType) {
+      case "addresses":
+        content = addresses.map(
+          ({ city, state, street, zip }: DataBrokerAddress, index: number) => (
+            <li key={`${props.dataBrokerDataType}-${index}`}>
+              {street}, {city}, {String(state)}, {zip}
```

Comment:
Q: Why does `state` need to be cast to `String()` but not the `street`/`city`/`zip`?